### DISCUSSION TITLE
How to return an image in a server API route?

### DISCUSSION THREAD

**Discussion by MAXOUXAX** [nuxt/nuxt#19735] [Questions] [answered]
Hey!
I'm using **Supabase Storage** and trying to **return an image from a server API route**, so that it displays correctly when opening the link in a new tab, or by setting the link as the `src` prop of an `img` tag.
The thing is, I don't know how to do it properly, since `await supabase.storage.from('bucket_name').download('file')` returns a Blob.

I've tried:
- directly returning the Blob (shows an empty JSON object `{}` even though the Blob contains data)
- returning the Blob and setting the `Content-Type` header to `image/jpeg` (shows an empty image, a gray square)
- converting the Blob to a base64 string (without the `Content-Type` header this just shows the string, with the header I get an empty image again)

Here's my current code:
```ts
const { data, error } = await supabase.storage.from('bucket_name').download('file');
if (!error) {
  setHeader(event, 'Content-Type', data.type)
  const result = await data.arrayBuffer();
  const img64 = Buffer.from(result).toString("base64");
  const img = `data:${data.type};base64, ${img64}`;
  return img;
}
// some other logic to return a default image
```

**Comment by MAXOUXAX** [comment, marked as answer]
I managed to fix the issue by doing this:
```ts
const { data, error } = await supabase.storage.from('bucket_name').download('file');
if (!error) {
  setHeader(event, 'Content-Type', data.type)
  const arrayBuffer = await data.arrayBuffer();
  const img64 = Buffer.from(arrayBuffer).toString("base64");
  return Buffer.from(img64, 'base64');
}
```

**Reply by bkan36** [reply]
Hi,
event is deprecated what I should use instead ? thanks
